File: multipicker/src/main/java/com/edwardstock/multipicker/picker/views/FilesView.java
package com.edwardstock.multipicker.picker.views;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.edwardstock.multipicker.data.MediaFile;
import com.edwardstock.multipicker.internal.views.ErrorView;

import java.io.File;

import androidx.recyclerview.selection.SelectionTracker;

/**
 * android-multipicker. 2018
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
public interface FilesView extends ErrorView, FileSystemView {
    void setAdapter(RecyclerView.Adapter<?> adapter);
    void setSelectionTitle(CharSequence title);
    void setOnSelectionClearListener(View.OnClickListener listener);
    void setOnSelectionDoneListener(View.OnClickListener listener);
    void setSelectionObserver(SelectionTracker.SelectionObserver<MediaFile> observer);
    void showEmpty();
    void startPreview(MediaFile file, View sharedView);
    void clearSelection();
    void finishWithResult();
    void showProgress();
    void hideProgress();
    void setSelectionSubmitEnabled(boolean enabled);
    void scrollTo(int position);
    void removeFileFromMediaDB(File file);
}

File: multipicker/src/main/java/com/edwardstock/multipicker/picker/views/FilesView.kt
package com.edwardstock.multipicker.picker.views

import android.view.View
import androidx.annotation.PluralsRes
import androidx.annotation.StringRes
import androidx.recyclerview.selection.SelectionTracker
import com.edwardstock.multipicker.data.MediaFile
import com.edwardstock.multipicker.internal.views.ErrorView
import java.io.File

/**
 * android-multipicker. 2018
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
interface FilesView : ErrorView, FileSystemView {
    fun setSelectionTitle(title: CharSequence)
    fun setSelectionTitle(@StringRes titleRes: Int)
    fun setSelectionTitleN(@PluralsRes titleRes: Int, n: Int)
    fun setSelectionTitlePhotosAndVideos(@PluralsRes prefix: Int, np: Int, @PluralsRes suffix: Int, ns: Int)
    fun setOnSelectionClearListener(listener: View.OnClickListener)
    fun setOnSelectionDoneListener(listener: View.OnClickListener)
    fun setSelectionObserver(observer: SelectionTracker.SelectionObserver<MediaFile>)
    fun startPreview(file: MediaFile, sharedView: View)
    fun clearSelection()
    fun finishWithResult()
    fun showProgress()
    fun hideProgress()
    fun setSelectionSubmitEnabled(enabled: Boolean)
    fun removeFileFromMediaDB(file: File)
}